### PR TITLE
feat(ds): Implement recalibration for sampling mode PROJECT

### PR DESCRIPTION
**Pull request by jan-auer**
Adds support for project mode in recalibration. In the top-level batch task, organizations are split in two groups:
- Organizations with a global sample rate. These go through the existing `recalibrate_orgs_batch` task, which internally now checks the configured target sample rate for the organization.
- Organizations in project mode. These go through a new `recalibrate_projects_batch`, which internally loads span volumes by project and stores a new recalibration cache entry for each of the projects.

Rule generation for this bias checks the sampling mode and loads and loads the project's recalibration factor in project mode.

Closes https://github.com/getsentry/projects/issues/193

### CODE REVIEW

File: src/sentry/dynamic_sampling/tasks/recalibrate_orgs.py [hunk 1]
```diff
@@ -141,3 +157,74 @@ def recalibrate_org(org_id: int, total: int, indexed: int) -> None:
...
     # At the end we set the adjusted factor.
     set_guarded_adjusted_factor(org_id, adjusted_factor)
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.recalibrate_projects_batch",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=25 * 60,
```
Comment (vgrozdanic) [inline src/sentry/dynamic_sampling/tasks/recalibrate_orgs.py:167]: `soft_time_limit` should be lower than `time_limit`

File: src/sentry/dynamic_sampling/rules/biases/recalibration_bias.py [hunk 2]
```diff
@@ -17,7 +21,11 @@ class RecalibrationBias(Bias):
     """
 
     def generate_rules(self, project: Project, base_sample_rate: float) -> list[PolymorphicRule]:
-        adjusted_factor = get_adjusted_factor(project.organization.id)
+        if is_project_mode_sampling(project.organization):
+            adjusted_factor = get_adjusted_project_factor(project.id)
+        else:
```
Comment (vgrozdanic) [inline src/sentry/dynamic_sampling/rules/biases/recalibration_bias.py:26]: nit: should we explicitly check if it is organization sampling mode?

**Review by vgrozdanic — APPROVED** [review]

File: src/sentry/dynamic_sampling/tasks/recalibrate_orgs.py (hunk 1 above)
Comment (jan-auer) [reply src/sentry/dynamic_sampling/tasks/recalibrate_orgs.py:167]: Agreed. I'll add this to our milestone, we need to clean up all the tasks and determine better time limits.

File: src/sentry/dynamic_sampling/rules/biases/recalibration_bias.py (hunk 2 above)
Comment (jan-auer) [reply src/sentry/dynamic_sampling/rules/biases/recalibration_bias.py:26]: Not in this case. This also applies to old organizations that were in neither mode.

**Pull request merged**